fix/restore-nicks.lua

```diff
@@ -0,0 +1,120 @@
+-- Workaround for the v50.x bug where Dwarf Fortress occasionally erase Dwarf's nicknames.
+-- It happen when killing certain figures, such as forgotten beasts.
+--@ enable = true
```

Comment:
you will also need `--@ module = true` so your script can automatically attach the state change handler and reenable itself after a load.

Reply:
Oh? It looked like it was correctly reenabled after restarting the save, but I may have missed something, thanks